src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -103,6 +103,16 @@ retry:
         // If the child and child_info link in the parent mismatch, we need to do btree repair, it might have
         // encountered a crash in-between the split or merge and only partial commit happened.
         if (is_split_needed(child_node, m_bt_cfg, req) || is_repair_needed(child_node, child_info)) {
+
+
+            if (!my_node->can_accomodate(m_bt_cfg, K::get_estimate_max_size(), BtreeLinkInfo::get_fixed_size())) {
+                // Mark the parent_node itself to be split upon next retry.
+                bt_thread_vars()->force_split_node = my_node;
+                unlock_node(child_node, child_cur_lock);
+                ret = btree_status_t::retry;
+                goto out;
+            }
```

Comment:
Why is this policy here? It is supposed to be in split_node

Reply:
As discussed, we want to call prepare node txn only if split node is successful.